examples/bri-3/src/bri/zkp/components/merkleTree/merkleTree.ts

```diff
@@ -1,4 +1,6 @@
 import { PreciseProofs } from 'ew-precise-proofs-js';
+import { Logger } from "tslog";
+const log: Logger = new Logger();
```

Comment:
i dont think we should instantiate new logger instance on all places where we use it, is there a way to instantiate and configure once and import everywhere where needed?

Reply:
@skosito each logger must be instantiated per app afaik. Unless there is a specific setting in the config file that I am not aware of. cc @Ybittan

Reply:
Removed unused logger. See LogConfig.ts and let me know if this is what you were hoping for

Reply:
I believe we should follow the NestJS modular approach here as well as logging is just another service that should be dependency injected. instead of instantiating it in a LogConfig.ts., could we create a LoggingModule that has a LoggingService registered and exported as a provider which can then be injected in any other module\agent\serveice?

Reply:
You can see how this was solved for[ winston](https://github.com/gremo/nest-winston) . Unfortunaltely ts-log [does not have this support for NestJs](https://github.com/fullstack-build/tslog/issues/89), but we can build something simple. For example:
LoggingModule would be in src/bri/shared/logging and LogggingService would be next to it, with the instantiation of the logger and interface exposing methods needed to perform the logs. LoggingService would be registered as provider in the LModule and the LoggingModule would be imported in the AppModule. This way we have a singleton logging service across the app, and we just inject where we need it.

Reply:
@ognjenkurtic yes agree, log config was some random example i found quickly on google but i agree it should be separate module that can be injected around
but also keep in mind we might have some files that are just plain classes or modules outside nestjs di (is this true?) that would also need to use this logger, for example merkle tree file here, but if its a module with exported methods should be enough

Reply:
if we want to be in nestjs world then why not this https://docs.nestjs.com/techniques/logger ?

Reply:
Based on the docs they dont seem too supportive to use it in production. I would stick with tslog inside of a logging module. 
We should not have a need for plain classes with business logic where logging is needed - business logic should always be in an agent or a service.

Reply:
not a business logic in this case but util class, so util service i presume in nestjs?
im fine with tslog, @Ybittan please let me know if you need some help with setting this up as @ognjenkurtic described

Reply:
Yes, @Ybittan please ping if you need any help with this.